// Copyright 2010-2021, Google Inc.
// All rights reserved.
//
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are
// met:
//
//     * Redistributions of source code must retain the above copyright
// notice, this list of conditions and the following disclaimer.
//     * Redistributions in binary form must reproduce the above
// copyright notice, this list of conditions and the following disclaimer
// in the documentation and/or other materials provided with the
// distribution.
//     * Neither the name of Google Inc. nor the names of its
// contributors may be used to endorse or promote products derived from
// this software without specific prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

syntax = "proto2";

package mozc.ibus;

message Config {
  repeated Engine engines = 1;
  // Whether IME is on (i.e. Hiragana mode) in the initial state.
  optional bool active_on_launch = 2;
}

message Engine {
  optional string name = 1;
  optional string longname = 2;
  optional string layout = 3;
  optional string layout_variant = 4;
  optional string layout_option = 5;
  // Greater numbers are prioritized more.
  optional int32 rank = 6 [default = 80];
  optional string symbol = 7 [default = "„ÅÇ"];
  enum CompositionMode {
    DIRECT = 0;
    HIRAGANA = 1;
    FULL_KATAKANA = 2;
    HALF_ASCII = 3;
    FULL_ASCII = 4;
    HALF_KATAKANA = 5;
    NONE = 100;
  }
  // Composition mode switched every time this engine is enabled.
  // The value NONE does not change the composition mode.
  optional CompositionMode composition_mode = 8 [default = NONE];
}
